generated/nifake/nifake/_converters.py

```diff
@@ -280,3 +281,27 @@ def convert_double_each_element(numbers):
     return [x * 2 for x in numbers]
 
 
+def convert_from_enum_with_converter_enum(value):
+    try:
+        return {
+            enums.EnumWithConverter.RED: True,
+            enums.EnumWithConverter.BLUE: False,
+            enums.EnumWithConverter.YELLOW: 'yellow',
+            enums.EnumWithConverter.BLACK: 42,
+        }[value]
+    except Exception:
+        raise ValueError(f"EnumWithConverter enum value '{value}' cannot be converted")
+
+
+def convert_to_enum_with_converter_enum(value):
+    try:
+        return {
+            True: enums.EnumWithConverter.RED,
+            False: enums.EnumWithConverter.BLUE,
+            'yellow': enums.EnumWithConverter.YELLOW,
+            42: enums.EnumWithConverter.BLACK,
+        }[value]
+    except Exception:
+        raise ValueError(f"'{value}' cannot be converted to an EnumWithConverter enum value")
```

Comment:
Under what circumstance would this occur?
I think that'd be when the driver returns an unexpected enum value.

When would that happen?
Maybe if the driver runtime is too new for the Python API. In which case `f"'{value}' cannot be converted to an EnumWithConverter enum value")` is useless to the customer. I think for this case we need a `DriverTooNewError` with a reasonable helpful message like:

> The driver runtime returned an unexpected value. This can occur if the `nidcpower` Python module is too old. Upgrade `nidcpower`.

Reply:
But perhaps this is not the place for that as this function doesn't have the context. This might be somethign to be changed in the Attribute class.

Reply:
I believe this comment applies to `convert_from_enum_with_converter_enum()` (the getter converter) instead of `convert_to_enum_with_converter_enum()` (the setter converter), since the value returned from the driver will only be passed to the getter converter.

I have created #1761 to add the new `DriverTooNewError`, and updated this PR to throw it from the `AttributeEnumWithConverter` attribute class (I have to throw it from there instead of the getter converter since the invalid return value from the driver will actually be caught by the [enum conversion there ](https://github.com/ni/nimi-python/pull/1738/files#diff-0ba5ea21d30086cc356b2093b1d46113f7bcc0c51c014c6b83115bde4f66b53fR127) before it gets passed into the getter converter)
```
            return self._getter_converter(
                self._underlying_enum_attribute.__get__(session, session_type)
            )
```